packages/shared-react/src/api/queries/transactionQueries.ts

```diff
@@ -249,3 +248,70 @@ export const nodeTransactionStatusQuery = ({ txHash, networkId, skip }: PendingT
         }
       : skipToken
   })
+
+interface NodeTransactionDecodeUnsignedTxQueryProps {
+  unsignedTx: string
+  networkId?: number
+  skip?: boolean
+}
+
+export const nodeTransactionDecodeUnsignedTxQuery = ({
+  unsignedTx,
+  networkId,
+  skip
+}: NodeTransactionDecodeUnsignedTxQueryProps) =>
+  queryOptions({
+    queryKey: ['transaction', 'node', 'decode', 'unsigned', unsignedTx],
+    ...getQueryConfig({ gcTime: ONE_MINUTE_MS, networkId }),
+    queryFn: !skip
+      ? async () => {
+          const result = await throttledClient.node.transactions.postTransactionsDecodeUnsignedTx({ unsignedTx })
+
+          const mockedTransaction: e.AcceptedTransaction = {
+            inputs: [],
+            outputs: result.unsignedTx.fixedOutputs.map((output) => ({ ...output, fixedOutput: true, type: '' })),
+            hash: result.unsignedTx.txId,
+            version: result.unsignedTx.version,
+            networkId: result.unsignedTx.networkId,
+            gasAmount: result.unsignedTx.gasAmount,
+            gasPrice: result.unsignedTx.gasPrice,
+            blockHash: '',
+            timestamp: 0,
+            scriptExecutionOk: true,
+            coinbase: false,
+            type: ''
+          }
+
+          for (const input of result.unsignedTx.inputs) {
+            const inputTxHash = await throttledClient.node.transactions.getTransactionsTxIdFromOutputref({
```

Comment:
I'm still a bit concerned about this. For 100+ inputs it's gonna be 100+ calls to the node, which are gonna be throttled and delayed a lot. 
I don't have a workaround in mind right now however.

Reply:
indeed, for 100 inputs, since we throttle to 10rps, we'd need min 10s.

@tdroxler mentioned that he could add a new EB endpoint and directly return inputs details from the output-ref.

I told him that it's not a high priority but def a good-to-have.

Reply:
This will still be 100 calls to EB :/

Reply:
ooooor there could be a nice beautiful new endpoint where we can submit the unsigned tx string and in return we'll get a tx object with full input and output details :) hehe